### DISCUSSION TITLE
METplus Level Error with ECMWF Obs Field (v11.0.2)

### DISCUSSION THREAD

**Discussion by ShannonShields-NOAA** [dtcenter/METplus#2180] [Configuration] [answered]
Hello,
I saw something similar to what I am about to ask in https://github.com/dtcenter/METplus/discussions/1066, but was hoping for a more specific/updated answer. I am trying to set name and level for ECMWF Grib1 data in my METplus GridStat config file. I have tried the following:
OBS_GRID_STAT_VAR1_NAME = 2T
OBS_GRID_STAT_VAR1_LEVELS = "(*,*)"
OBS_GRID_STAT_VAR1_LEVELS = (*,*)
OBS_GRID_STAT_VAR1_LEVELS = '(*,*)'
However, I keep receiving the error message:
ERROR  : VarInfo::set_level_info_grib() - failed to parse level string '(*,*)'
So I was looking on this discussions forum and it seems I need to be setting grib table info to tell METplus how to search for the 2m temp variable. I was wondering what table info I need to set/how for ECMWF files. I have copied an example file onto Hera:
/scratch2/NCEPDEV/stmp1/Shannon.Shields/data/ecmwf.2023051600.anl
When I wgrib ecmwf.2023051600.anl and search for 2m temp, I get:
20:1696212:d=23051600:2T:kpds5=167:kpds6=1:kpds7=0:TR=0:P1=0:P2=0:TimeU=1:sfc:anl:type=analysis:NAve=0

**Comment by ShannonShields-NOAA** [comment]
I'm not sure why the star symbols didn't show up in the parentheses when I posted, but two stars separated by a comma representing lat and lon are what is set in my config file.

**Comment by j-opatz** [comment, marked as answer]
Hi Shannon,

And thank you for your question. No worries on the missing asterisks, it's a common problem in text editors. Asterisks are sometimes used to create italics, with anything between two asterisks becoming italicized. Putting a backslash (\) character in front of the asterisk usually cancels this behavior, ex. \*,* .

As for your indicated problem, the parentheses and asterisks are used only for netCDF input levels. For grib1 input files, you'll need to use the desired standard level or pressure level as you did with the MET configuration file. [Here's a link to the MET User's Guide section that discusses this behavior](https://met.readthedocs.io/en/develop/Users_Guide/config_options.html#fcst), you'll need to scroll down a bit to the "file format" section. To grab the important bit out for you,
 
```
 The “level” entry specifies a level type and value:

    ANNN for accumulation interval NNN
    ZNNN for vertical level NNN
    ZNNN-NNN for a range of vertical levels
    PNNN for pressure level NNN in hPa
    PNNN-NNN for a range of pressure levels in hPa
    LNNN for a generic level type
    RNNN for a specific GRIB record number
```

This behavior extends to the METplus wrapper configuration files as well. So for example, if you wanted to pull out the 2T variable at the first standard vertical level output, you'd enter:

OBS_GRID_STAT_VAR1_NAME = 2T
OBS_GRID_STAT_VAR1_LEVELS = Z0

I hope that helps you out!

**Reply by ShannonShields-NOAA** [reply]
Ah, here I was thinking it was more complicated than simply setting "OBS_GRID_STAT_VAR1_LEVELS = Z0". Thank you @j-opatz ; that worked! However, I now have another problem in my grid_stat*.nc output files...the forecast, obs, and climo mean variable names do not match:
FCST_TMP_Z2_FULL
OBS_2T_Z0_FULL
CLIMO_MEAN_2T_Z0_FULL
Can I make the OBS and CLIMO_MEAN match FCST by setting something like the following in my METplus config file?
OBS_VAR1_OPTIONS = set_attr_name = "OBS_TMP_Z2_FULL";
GRID_STAT_CLIMO_MEAN_VAR1_OPTIONS = set_attr_name = "CLIMO_MEAN_TMP_Z2_FULL";

**Reply by ShannonShields-NOAA** [reply]
I figured out how to correct the variable name and level for the OBS and CLIMO_MEAN output! I had to set:
OBS_VAR1_OPTIONS = set_attr_name = "TMP"; set_attr_level = "Z2";
GRID_STAT_CLIMO_MEAN_VAR1_OPTIONS = set_attr_name = "TMP"; set_attr_level = "Z2";
Now my output .nc files are matching and I can run my next steps to get my final stat file.
